nebula/ui/components/VPNMenu.qml

```diff
@@ -29,57 +29,76 @@ Item {
     Rectangle {
         id: menuBackground
         color: VPNTheme.theme.bgColor
-        y: 0
-        width: parent.width
-        height: 55
+        anchors.fill: parent
     }
 
-    VPNIconButton {
-        id: iconButton
-
-        skipEnsureVisible: true // prevents scrolling of lists when this is focused
-
-        onClicked: _menuOnBackClicked()
-        anchors.top: parent.top
-        anchors.left: parent.left
-        anchors.topMargin: VPNTheme.theme.windowMargin / 2
-        anchors.leftMargin: VPNTheme.theme.windowMargin / 2
-        //% "Back"
-        //: Go back
-        accessibleName: qsTrId("vpn.main.back")
-        Accessible.ignored: accessibleIgnored
-
-        enabled: !btnDisabled
-        opacity: enabled ? 1 : .4
-        Image {
-            objectName: "menuIcon"
-            source: _iconButtonSource
-            sourceSize.width: VPNTheme.theme.iconSize
-            fillMode: Image.PreserveAspectFit
-            anchors.centerIn: iconButton
+    RowLayout {
+        id: row
+        anchors {
+            fill: parent
+            margins: VPNTheme.theme.windowMargin / 2
+        }
+        spacing: VPNTheme.theme.windowMargin / 4
+
+        VPNIconButton {
+            id: iconButton
+
+            skipEnsureVisible: true // prevents scrolling of lists when this is focused
+
+            onClicked: _menuOnBackClicked()
+            Layout.alignment: Qt.AlignLeft
+            //% "Back"
+            //: Go back
+            accessibleName: qsTrId("vpn.main.back")
+            Accessible.ignored: accessibleIgnored
+            Layout.preferredHeight: VPNTheme.theme.rowHeight
+            Layout.preferredWidth: VPNTheme.theme.rowHeight
+            enabled: !btnDisabled
+            opacity: enabled ? 1 : .4
+
+            Image {
+                objectName: "menuIcon"
+                source: _iconButtonSource
+                sourceSize.width: VPNTheme.theme.iconSize
+                fillMode: Image.PreserveAspectFit
+                anchors.centerIn: iconButton
+            }
```

Comment:
Suggestion: Definitely not part of this PR, but maybe we could file a ticket under tech debt — Looking at the `VPNIconButton` and places where it’s currently used I think it would be nice if `Image` could be moved into the component, so that we can just pass the `source` property.